src/operation-node/query-node.ts

```diff
@@ -43,6 +44,18 @@ export const QueryNode = freeze({
     )
   },
 
+  cloneWithEndModifier<T extends HasEndModifiers>(
+    node: T,
+    modifier: OperationNode,
+  ): T {
+    return freeze({
+      ...node,
+      endModifiers: query.endModifiers
```

Comment:
missed that one.. sorry.

```suggestion
      endModifiers: node.endModifiers
```